main.tf

```diff
@@ -47,14 +60,18 @@ resource "aws_internet_gateway" "main" {
   vpc_id = aws_vpc.main.id
 
   tags = {
-    Name = "main"
+    Name = "main${aws_vpc.main.id}"

```

Comment:
Instead of `aws_vpc.main.id` please introduce `var.env_code` and interpolate it in all Name tags both here and in the rest of the code. Please create separate variables.tf file and declare all your variables there